Eastern-Research-Group/mywaterway
# Feature/upstream watersheds widget

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3470390

## Main Changes:
* Adds new widget to Community map for displaying the entire upstream watershed of the current huc12
* errorMessage moved into locationSearch context so that it is accessible through Upstream Widget and future widgets.
* Per client request, the Upstream Layer appears in the Legend widget but does **not** appear in the Layer List widget

## Steps To Test:
1. Navigate to http://localhost:3000/community
2. Search a location. Good locations are "DC", "Boston", "080701000104"
3. Click the Upstream Watershed widget in the top right corner of the map. Once the fetch completes the map should zoom to the full upstream extent.
4. Clicking the widget again should zoom back to the current extent.
5. Click the Upstream Layer in a place that **is not** within the huc12 boundaries. A popup with the area and ability to change location should appear.
6 Click the Upstream Layer in a place that **is** within the Huc12 boundaries. A popup with just the area should appear.


Notes: 
- upstream watershed layer color is currently pending and will be changed at a later time, likely after this branch is merged.
- Alaska is not currently supported for this service
